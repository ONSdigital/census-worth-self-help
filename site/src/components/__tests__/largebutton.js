import React from "react"
import renderer from "react-test-renderer"
import { render, fireEvent } from "react-testing-library"

import LargeButton from "../largebutton"

import { navigate } from "@reach/router"

jest.mock('@reach/router', () => ({
  navigate: jest.fn(),
}))

describe("Large Button", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(<LargeButton title="press me" link="/anywhere"/>)
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it("navigates when clicked", () => {
  	const { getByTestId } = render(<LargeButton title="press me" link="/anywhere"/>);
  	
    const button = getByTestId('large-button')

  	expect(button.textContent).toEqual("press me")

  	fireEvent.click(button);

  	expect(navigate).toHaveBeenCalledTimes(1)
    expect(navigate).toHaveBeenCalledWith('/anywhere')
  })
})