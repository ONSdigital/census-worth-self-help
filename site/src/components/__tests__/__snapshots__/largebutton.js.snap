// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Large Button renders correctly 1`] = `
<div
  css={
    Object {
      "map": undefined,
      "name": "17k9f9t",
      "next": undefined,
      "styles": "
        
    display: flex;
    justify-content: center;
    flex-direction: column;
  
        padding: 0px 10px;
        border-radius: 4px;
        box-shadow: 0 2px 4px 0 rgba(212, 212, 212, 0.5);
        border: solid 1px #ececed;
        margin: 0px 15px;
        text-align: center;
        height: 50px;
        background-color: rgb(255, 255, 255);
        
      ",
    }
  }
  data-testid="large-button"
  onClick={[Function]}
>
  <div
    className="Button-heading-Style"
    css={
      Object {
        "map": undefined,
        "name": "7tk8ru",
        "next": undefined,
        "styles": "
          margin: 0px 20px;
          color: rgb(0,61,89);
        ",
      }
    }
  >
    press me
  </div>
</div>
`;
